storage/badger/operation/protocol_kv_store.go

```diff
@@ -0,0 +1,40 @@
+package operation
+
+import (
+	"github.com/dgraph-io/badger/v2"
+
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/storage"
+)
+
+// InsertProtocolKVStore inserts a protocol KV store by ID.
+// Error returns:
+//   - storage.ErrAlreadyExists if the key already exists in the database.
+//   - generic error in case of unexpected failure from the database layer or encoding failure.
+func InsertProtocolKVStore(protocolKVStoreID flow.Identifier, kvStore *storage.KeyValueStoreData) func(*badger.Txn) error {
+	return insert(makePrefix(codeProtocolKVStore, protocolKVStoreID), kvStore)
+}
+
+// RetrieveProtocolKVStore retrieves a protocol KV store by ID.
+// Error returns:
+//   - storage.ErrNotFound if the key does not exist in the database
+//   - generic error in case of unexpected failure from the database layer
+func RetrieveProtocolKVStore(protocolKVStoreID flow.Identifier, kvStore *storage.KeyValueStoreData) func(*badger.Txn) error {
+	return retrieve(makePrefix(codeProtocolKVStore, protocolKVStoreID), kvStore)
+}
+
+// IndexProtocolKVStore indexes a protocol KV store by block ID.
+// Error returns:
+//   - storage.ErrAlreadyExists if the key already exists in the database.
+//   - generic error in case of unexpected failure from the database layer or encoding failure.
```

Comment:
don't think we encode much here:
```suggestion
//   - generic error in case of unexpected failure from the database layer
```